packages/@react-spectrum/menu/docs/types.ts

```diff
@@ -0,0 +1,9 @@
+import {ReactElement} from 'react';
+
+// The doc generator is not smart enough to handle the real types for ContextualHelpTrigger so this is a simpler one.
+export interface ContextualHelpTriggerProps {
+  /** Whether the menu item is currently unavailable. */
+  isUnavailable?: boolean,
+  /** The Item which triggers opening of the Dialog, and the Dialog itself. */
+  children: [ReactElement, ReactElement]
+}
```

Comment:
Sorry about the back and forth with this, but perhaps we should move this interface definition into ContextualHelpTrigger and set it as the SpectrumMenuDialogTriggerProps so the exported props match the actual props the user would provide? Also we'd be able to use it directly as the `let _Item = ContextualHelpTrigger...` props and thus wouldn't risk drift between this doc definition here and the `Pick` set in ContextualHelpTrigger?

aka:

```
// In packages/@react-spectrum/menu/src/ContextualHelpTrigger.tsx
interface MenuDialogTriggerProps<T> extends ItemProps<T> {
  isUnavailable?: boolean,
  targetKey: Key
}

export interface SpectrumMenuDialogTriggerProps extends Pick<MenuDialogTriggerProps<T>, 'isUnavailable' | 'children'>) {}

function ContextualHelpTrigger<T>(props: MenuDialogTriggerProps<T>): ReactElement {
...

let _Item = ContextualHelpTrigger as <T>(props: SpectrumMenuDialogTriggerProps<T>) => JSX.Element;
```
and the mdx file could do something like:

```
import contextualHelpTriggerTypes from 'docs:@react-spectrum/menu/src/ContextualHelpTrigger.tsx';

// Similar to how we import Column props in TableView.mdx
<PropTable component={{
  props: {
    properties: Object.fromEntries(Object.entries(menu.exports.SpectrumMenuDialogTriggerProps.properties))
  }
}} links={contextualHelpTriggerTypes.links} style={{marginBottom: '40px'}} />

```

I forget if there is an easier way to grab the types here, maybe `docs.exports.SpectrumMenuDialogTriggerProps` works?

Reply:
I think I managed to get it to work. When using Pick I wasn't seeing anything in the props table, so I updated to use Omit instead and rearranged some things.